reflex/state.py

```diff
@@ -696,6 +698,19 @@ def _check_overridden_methods(cls):
                 f"The event handler name `{method_name}` shadows a builtin State method; use a different name instead"
             )
 
+    @classmethod
+    def _check_overriden_basevars(cls):
+        """Check for shadow base vars and raise error if any.
+
+        Raises:
+            NameError: When a computed var shadows a base var.
+        """
+        for name, value in cls.__dict__.items():
```

Comment:
how about just using `_get_computed_vars` instead? It will check for conflicts in mixins as well